### PR TITLE
[READY TO MERGE] Adding workspace level quota management APIs in version 2019-06-01

### PR DESCRIPTION
**Pull request by leizhaoPS**
This feature related changes is in **https://github.com/Azure/azure-rest-api-specs/pull/6366/commits/4ac4d6ae954a2ce2891969e8c617e15efdaa33e7**
To update quota, we use the Post call with the action quota URI (similar to Usages): subscriptions/{subscriptionId}/providers/Microsoft.BatchAI/locations/{locationName}/updateQuotas

Parameters of this call:
Subscription ID (uri)
location (uri)
vmFamily (body)
workspaceId (body)
Quota (body)
api-version (query)

To get quota, we use Get call on the quota URI: subscriptions/{subscriptionId}/providers/Microsoft.BatchAI/locations/{locationName}/listQuotas

Parameters of this call:
Subscription ID (uri)
location (uri)
api-version (query)
Changes:
Added new API for update Quotas
Added new API for list Quotas
Added two examples:
ListWorkspaceQuotasByVMFamily.json
UpdateQuota.json

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by erich-wang** [comment]
@leizhaoPS , please fix CI errors.

**Review by erich-wang — CHANGES_REQUESTED** [review]
Please fix CI errors

**Comment by leizhaoPS** [comment]
@erich-wang , could you please help take a look at the design part related to workspace level quota managment API: https://github.com/Azure/azure-rest-api-specs/commit/4ac4d6ae954a2ce2891969e8c617e15efdaa33e7, other validation failures we already fixed in the main branch.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 1]
```diff
@@ -0,0 +1,2535 @@
...
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/MachineLearningServiceError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Create a cluster": {
+            "$ref": "./examples/UpdateQuota.json"
```
Comment (nschonni) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: Think the casing mismatch is the issue with semantic
```suggestion
            "$ref": "./examples/updateQuota.json"
```

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 2]
```diff
@@ -0,0 +1,2535 @@
...
```
Comment (nschonni) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]:
```suggestion
            "$ref": "./examples/updateQuota.json"
```

**Comment by leizhaoPS** [comment]
@erich-wang, I fixed all the checks issues and merged master to this branch, now only feature related changes need review, please help take a look at those changes.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 3]
```diff
@@ -1248,6 +1343,156 @@
...
+          "readOnly": true,
+          "type": "string",
+          "description": "Specifies the resource type."
+        },
+        "name": {
+          "$ref": "#/definitions/ResourceName",
+          "description": "Name of the resource.",
+          "readOnly": true
+        },
+        "quota": {
```
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: prefer "limit"

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 4]
```diff
@@ -1248,6 +1343,156 @@
...
+      "description": "The result of update workspace quota."
+    },
+    "UpdateWorkspaceQuotas": {
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Specifies the resource ID."
+        },
+        "type": {
```
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: should this be an enum? How many types are supported?

**Review by KrisBash — CHANGES_REQUESTED** [review]
left a few comments

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 5]
```diff
@@ -442,6 +442,101 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.MachineLearningServices/locations/{location}/updateQuotas": {
```
[the remark this answers is not in the document]
Comment (leizhaoPS) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: Agree

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json (hunk 4 above)
Comment (leizhaoPS) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: for "id" and "type", we will follow the same as usages contract: https://microsoft-my.sharepoint.com/:w:/p/gauravbh/EYPNGVUmC3hGskyHRaTk4cABlwkm6ziCMnmqLMj0QSJ0Vg?rtime=BW9QFs311kg

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 6]
```diff
@@ -442,6 +442,101 @@
...
+            "description": "Quota update parameters."
+          },
+          {
+            "$ref": "#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```
[the remark this answers is not in the document]
Comment (leizhaoPS) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: this is await call and FE will wait for the response with status.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json [hunk 7]
```diff
@@ -442,6 +442,101 @@
...
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Create a cluster": {
+            "$ref": "./examples/updateQuota.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.MachineLearningServices/locations/{location}/Quotas": {
```
[the remark this answers is not in the document]
Comment (leizhaoPS) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: Same as above, I feel it express the same as limits: "Quota": https://dictionary.cambridge.org/us/dictionary/english/quota: a fixed, limited amount or number that is officially allowed

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (leizhaoPS) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: we may want use quotas to keep the name consistent with the sub level quota concept

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (KrisBash) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: This is marked with x-ms-long-running-operation: true, but does not have the correct response code modeled (201).

**Comment by erich-wang** [comment]
@KrisBash, please change the label to **ARMSignOff** if you don't have any other concern, thanks.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json (hunk 3 above)
Comment (leizhaoPS) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-05-01/machineLearningServices.json]: I changed from "quota" to "limit"

**Comment by ravbhatnagar** [comment]
@KrisBash - please let me know if you are good to sign off on this.

**Review by KrisBash — APPROVED** [review]
thanks for the changes

**Review by erich-wang — APPROVED** [review]

**Comment by erich-wang** [comment]
@leizhaoPS , please let me know when PR is ready for merge.

**Comment by leizhaoPS** [comment]
Thanks for follow up! I will check with the team when backend rollout and let you know.:)

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by erich-wang** [comment]
@leizhaoPS , any process for this PR?

**Comment by erich-wang** [comment]
@@leizhaoPS , I added **DoNotMerge** for this PR, please let me know when it is ready for merge.

**Comment by yungezz** [comment]
Hi @leizhaoPS what's your plan to proceed this PR? is it ok to close it for now, pls reopen or create a new PR when service is ready

**Review by raych1 — APPROVED** [review]

**Pull request merged**